src/v/archival/service.cc

```diff
@@ -397,6 +397,16 @@ scheduler_service_impl::remove_archivers(std::vector<model::ntp> to_remove) {
       .finally([g = std::move(g)] {});
 }
 
+std::optional<model::offset>
+scheduler_service_impl::get_high_watermark(const model::ntp& ntp) const {
+    cluster::partition_manager& pm = _partition_manager.local();
+    auto p = pm.get(ntp);
+    if (p) {
+        return p->high_watermark();
```

Comment:
do you want to use `high_watermark` or `last_stable_offset`? Maybe @mmaslankaprv can explain the difference, i'm not entirely sure of the difference myself

Reply:
updated the commit message